### PR TITLE
TTT: Reintroduction of the possibility to take DNA from players

### PR DESCRIPTION
**Pull request by markusmarkusz**
I last saw when I was modifing the DNA scanner for me, that there was "self: GatherPlayerSample" from an older version.
And I thought, "Why not just re-implement this function?".
Because I have to much time at the moment, I have just done this.
If someone want this back this can be merged.
If not then maybe you should remove the rests from this function :p.

### CODE REVIEW

File: garrysmod/gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua [hunk 1]
```diff
@@ -107,10 +110,10 @@ function SWEP:PrimaryAttack()
 
    local tr = util.TraceLine({start=spos, endpos=sdest, filter=self.Owner, mask=MASK_SHOT})
    local ent = tr.Entity
-   if IsValid(ent) and (not ent:IsPlayer()) then
+   if IsValid(ent) then
       if SERVER then
-         if ent:IsPlayer() then
-            --self:GatherPlayerSample(ent)
+         if ent:IsPlayer() and scanplayer then
```
Comment (bmwalters) [inline garrysmod/gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua]: should be scanplayer:GetBool(), did you test this before creating a pull request?

File: garrysmod/gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua [hunk 2]
```diff
@@ -388,7 +408,7 @@ if CLIENT then
              ent:GetNWBool("HasPrints", false) or
              (ent:GetClass() == "prop_ragdoll" and
               CORPSE.GetPlayerNick(ent, false) and
-              CORPSE.GetFound(ent, false))) then
+              CORPSE.GetFound(ent, false)) or (ent:IsPlayer() and scanplayer)) then
```
Comment (bmwalters) [inline garrysmod/gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua]: Again, should be `scanplayer:GetBool()`

**Comment by bmwalters** [comment]
Gameplay-wise this could be a bit over-powered, right? A cheeky detective could grab DNA of everyone at the beginning of the round and have the ability to track down individuals without the help of radar.

File: garrysmod/gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua (hunk 1 above)
Comment (markusmarkusz) [inline garrysmod/gamemodes/terrortown/entities/weapons/weapon_ttt_wtester.lua]: Yeah, but I didn't use the ConVar for testing.
I added it after testing the function...
Changed it.

**Comment by bmwalters** [comment]
Quick tip; while your work is under code review, it's nice to not squash commits in order to see what has been changed between revisions.

**Comment by markusmarkusz** [comment]
@bmwalters 
Okay. I will do this from now on. q:
I only re-added this function. So if it is to over-powered maybe I can make it weak.

**Comment by svdm** [comment]
The concept of grabbing DNA from players comes from an earlier version of the DNA scanner that worked in a completely different way. Detectives had to grab player's DNA and compare it to DNA found on items and corpses. This might sound interesting but was too complex in practice, so nobody bothered to use it. As a result I greatly simplified the DNA scanner into something closer to what we have now.

There's no point in taking the current DNA scanner mechanics and only reintroducing the ability to grab DNA from players. The game is not balanced for that at all.

In general, pull requests are not the place to make significant gameplay changes/additions. The core gameplay stuff of TTT is not going to change anymore at this point (unless someday I decide to spend serious time on designing, implementing and testing such gameplay change again). That's what custom weapons, workshop mods, etc are for.

**Comment by markusmarkusz** [comment]
@svdm
Should I make a pull request for the rests of this function?
For example that this self:GatherPlayerSample will be removed?

**Comment by svdm** [comment]
Sure, go ahead.

**Pull request closed without merging**